nautobot_circuit_maintenance/views.py

```diff
@@ -189,22 +199,19 @@ def get_maintenances_per_month(self):
         return self.queryset.count() / delta_months
 
 
-class CircuitMaintenanceListView(generic.ObjectListView):
-    """View for listing the config circuitmaintenance feature definition."""
+class CircuitMaintenanceUIViewSet(NautobotUIViewSet):
+    """UIViewSet for CircuitMaintenance."""
 
+    bulk_update_form_class = forms.CircuitMaintenanceBulkEditForm
+    filterset_class = filters.CircuitMaintenanceFilterSet
+    filterset_form_class = forms.CircuitMaintenanceFilterForm
+    form_class = forms.CircuitMaintenanceForm
     queryset = models.CircuitMaintenance.objects.order_by("-start_time")
-    table = tables.CircuitMaintenanceTable
-    filterset = filters.CircuitMaintenanceFilterSet
-    filterset_form = forms.CircuitMaintenanceFilterForm
+    serializer_class = serializers.CircuitMaintenanceSerializer
+    table_class = tables.CircuitMaintenanceTable
     action_buttons = ("add", "export")
 
-
-class CircuitMaintenanceView(generic.ObjectView):
-    """Detail view for specific circuit maintenances."""
-
-    queryset = models.CircuitMaintenance.objects.all()
-
-    def get_extra_context(self, request, instance):
+    def get_extra_context(self, request, instance=None):
         """Extend content of detailed view for Circuit Maintenance."""
         maintenance_note = models.Note.objects.filter(maintenance=instance)
```

Comment:
Are these making unnecessary database calls on non-detail views? Maybe we should put these in a `if instance is not None:` condition?

Reply:
I think a cleaner option may be
```py
if instance is None:
    return {}
```